devicetypes/timevalve-gaslock-t-08/timevalve-smart.src/timevalve-smart.groovy

```diff
@@ -0,0 +1,281 @@
+    	        }
+                tileAttribute("device.remainingText", key: "SECONDARY_CONTROL") {
+	            	attributeState "open", label: '${currentValue}', icon:"st.contact.contact.open", backgroundColor:"#ffa81e"
+	            	attributeState "closed", label:'', icon:"st.contact.contact.closed", backgroundColor:"#79b821"
+    	        }
+            }
+            
+            standardTile("refreshTile", "command.refresh", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+                state "default", label:'', action:"refresh.refresh", icon:"st.secondary.refresh"
+            }
+            
+            controlTile("remainingSliderTile", "device.remaining", "slider", inactiveLabel: false, range:"(0..590)", height: 2, width: 4) {
+            	state "level", action:"remainingLevel"
+            }
+            valueTile("setRemaining", "device.remainingText", inactiveLabel: false, decoration: "flat", height: 2, width: 2){
+                state "remainingText", label:'${currentValue}\nRemaining'//, action: "setRemaining"//, icon: "st.Office.office6"
+            }
+
+            standardTile("setTimeout10", "device.remaining", inactiveLabel: false, decoration: "flat") {
+                state "default", label:'10Min', action: "setTimeout10", icon:"st.Health & Wellness.health7", defaultState: true
+                state "10", label:'10Min', action: "setTimeout10", icon:"st.Office.office13"
+            }
+            standardTile("setTimeout20", "device.remaining", inactiveLabel: false, decoration: "flat") {
+                state "default", label:'20Min', action: "setTimeout20", icon:"st.Health & Wellness.health7", defaultState: true
+                state "20", label:'20Min', action: "setTimeout20", icon:"st.Office.office13"
+            }
+            standardTile("setTimeout30", "device.remaining", inactiveLabel: false, decoration: "flat") {
+                state "default", label:'30Min', action: "setTimeout30", icon:"st.Health & Wellness.health7", defaultState: true
+                state "30", label:'30Min', action: "setTimeout30", icon:"st.Office.office13"
+            }
+            standardTile("setTimeout40", "device.remaining", inactiveLabel: false, decoration: "flat") {
+                state "default", label:'40Min', action: "setTimeout40", icon:"st.Health & Wellness.health7", defaultState: true
+                state "40", label:'40Min', action: "setTimeout40", icon:"st.Office.office13"
+            }
+
+            valueTile("batteryTile", "device.battery", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+                state "battery", label:'${currentValue}% battery', unit:""
+            }
+
+            main (["statusTile"])
+//            details (["statusTile", "remainingSliderTile", "setRemaining", "setTimeout10", "setTimeout20", "batteryTile", "refreshTile", "setTimeout30", "setTimeout40"])
+//            details (["statusTile", "batteryTile", "setRemaining", "refreshTile"])
```

Comment:
yes. setRemaining never called.
several function and tile hide by intention.
function is operating well.
However, it not reached the UI requirements.